accounts/abi/bind/backends/simulated_test.go

```diff
@@ -1336,3 +1339,195 @@ func TestForkResendTx(t *testing.T) {
 		t.Errorf("TX included in wrong block: %d", h)
 	}
 }
+
+// TestLowGasTxNotBeingMined checks that the lower gas fee tx with same nonce does not get mined
+// Steps:
+// 1. Send a tx lower than the set market gas price
+// 2. Commit to chain, check nonce stays the same
+// 3. Send a tx meets the market gas price with same nonce.
+// 4. Commit to chain
+// 5. Check tx2 is not in pending state and nonce has increased
+func TestLowGasTxNotBeingMined(t *testing.T) {
+	testAddr := crypto.PubkeyToAddress(testKey.PublicKey)
+	var gas uint64 = 21000
+
+	sim := simTestBackend(testAddr)
+	defer sim.Close()
+
+	var testCases = []struct {
+		name     string
+		txType	 byte
+	}{
+		{
+			name: "LegacyTx",
+			txType: types.LegacyTxType,
+		},
+		{
+			name: "dynamicTx",
+			txType: types.DynamicFeeTxType,
+		},
+	}
+	for i, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			bgCtx := context.Background()
+			// Create tx and send
+			head, _ := sim.HeaderByNumber(context.Background(), nil) // Should be child's, good enough
+			gasPrice := new(big.Int).Add(head.BaseFee, big.NewInt(1))
```

Comment:
nit: can use get suggested fee tip method I believe instead of just hard coding 1